android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModule.java

```diff
@@ -1,22 +1,320 @@
 
 package com.reactnativecommunity.webview;
 
+import android.Manifest;
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Build;
+import android.provider.MediaStore;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.FileProvider;
+import android.util.Log;
+import android.webkit.ValueCallback;
+import android.webkit.WebChromeClient;
+
+import com.facebook.react.ReactActivity;
+import com.facebook.react.bridge.ActivityEventListener;
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.ReactContextBaseJavaModule;
-import com.facebook.react.bridge.ReactMethod;
-import com.facebook.react.bridge.Callback;
+import com.facebook.react.modules.core.PermissionAwareActivity;
+import com.facebook.react.modules.core.PermissionListener;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
 
-public class RNCWebViewModule extends ReactContextBaseJavaModule {
+import static android.app.Activity.RESULT_OK;
+
+public class RNCWebViewModule extends ReactContextBaseJavaModule implements ActivityEventListener {
 
   private final ReactApplicationContext reactContext;
+  private RNCWebViewPackage aPackage;
+
+  private static final int REQUEST_CAMERA = 1;
+  private static final int SELECT_FILE = 2;
+
+  private ValueCallback<Uri[]> filePathCallback;
+  private Uri outputFileUri;
+  private String intentTypeAfterPermissionGranted;
+
+  // @todo this could be configured from JS
```

Comment:
When you say "this could be configured from JS" it's from RN app code or from JS executed with HTML in WebView?

IHMO that just from RN app code will be enough to this kind of personalizations and translations strings.

What do you think?

Reply:
Sorry for the long silence @lucasferreira 
I was thinking of the same thing, to be passed from the RN app.

Regarding the default accepted types, I would default it to `*/*` in the Java Module, so you could select any type of file, including: pdf/doc/etc (similar to how web works). You should be able override this using the `accept` attribute from the web page.

Regarding the translations for strings, one solution would be to be able to override them from RN, passed as props.